File: works/tmp/1105.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main
{
    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] _s = br.readLine().split(" ");
        String n = _s[0];
        String m = _s[1];

        while(n.length() != m.length())
            n = "0" + n;

        int cnt = 0;
        String _n = n;
        String _m = m;
        for(int i = n.length()-1; i >= 0; i--)
        {
            if(n.charAt(i) == '8' && m.charAt(i) == '8')
            {
                n = n.substring(0,i) + '9' + n.substring(i+1);
                if(n.compareTo(m) > 0) cnt++;
                else { n = _n; m = _m; }
            }
        }
        System.out.println(cnt);
    }

}

File: works/tmp/1105.kt
import java.util.*
import java.io.*;
import java.io.BufferedWriter;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;


fun main()
{
    val br = BufferedReader(InputStreamReader(System.`in`))
    var _s = br.readLine().split(' ')
    var n = _s[0]
    var m = _s[1]

    while(n.length != m.length)
        n = "0" + n

    var cnt = 0
    var _n = n
    var _m = m
    for(i in n.length-1 downTo  0)
    {
        if(n[i] == '8' && m[i] == '8')
        {
            n = n.substring(0,i) + '9' + n.substring(i+1)
            if(n.compareTo(m) > 0) cnt++
            else { n = _n; m = _m }
        }
    }
    print(cnt)
}